src/draw/lv_img_decoder.h

```diff
@@ -28,17 +29,44 @@ extern "C" {
 /**********************
  *      TYPEDEFS
  **********************/
+#if defined(LV_IMG_FRAME_INDEX_LARGE_SIZE)
+typedef uint32_t lv_frame_index_t;
+#else
+typedef uint16_t lv_frame_index_t;
+#endif
 
 /**
- * Source of image.*/
-enum {
-    LV_IMG_SRC_VARIABLE, /** Binary/C variable*/
-    LV_IMG_SRC_FILE, /** File in filesystem*/
-    LV_IMG_SRC_SYMBOL, /** Symbol (@ref lv_symbol_def.h)*/
-    LV_IMG_SRC_UNKNOWN, /** Unknown source*/
-};
+ * Capabilities of an image decoder context.
+ * Set by the decoder when extracting info or being opened */
+typedef enum {
+    LV_IMG_DEC_DEFAULT      = 0x00, /**!< Default format, no specificities */
+    LV_IMG_DEC_VECTOR       = 0x01, /**!< Image format is vector based, size independant */
+    LV_IMG_DEC_ANIMATED     = 0x02, /**!< Image format stores an animation */
+    LV_IMG_DEC_SEEKABLE     = 0x04, /**!< Animation is seekable */
+    LV_IMG_DEC_CACHED       = 0x08, /**!< The complete image can be cached (used for rotation and zoom) */
+    LV_IMG_DEC_VFR          = 0x10, /**!< The animation has a variable frame rate */
+    LV_IMG_DEC_LOOPING      = 0x20, /**!< The animation is looping */
+} lv_img_decoder_caps_t;
```

Comment:
```suggestion
} lv_img_dec_caps_t;
```

We used `dec` on other places too.